examples/ltimes.cpp

```diff
@@ -111,751 +113,808 @@ int main(int RAJA_UNUSED_ARG(argc), char **RAJA_UNUSED_ARG(argv[]))
-
-  std::array<RAJA::idx_t, 3> psi_perm {{0, 1, 2}};
-  PsiView psi(psi_data,
-              RAJA::make_permuted_layout({{num_d, num_g, num_z}}, psi_perm));
-
-  std::array<RAJA::idx_t, 3> phi_perm {{0, 1, 2}};
-  PhiView phi(phi_data,
-              RAJA::make_permuted_layout({{num_m, num_g, num_z}}, phi_perm));
-
-  using EXECPOL = 
-    RAJA::KernelPolicy<
-       statement::For<0, loop_exec,  // m       
-         statement::For<1, loop_exec,  // d
-           statement::For<2, loop_exec,  // g 
-             statement::For<3, simd_exec,  // z
-               statement::Lambda<0>
-             >
-           >
-         >
-       >
-     >;
-
-  auto segments = RAJA::make_tuple(RAJA::TypedRangeSegment<IM>(0, num_m),
-                                   RAJA::TypedRangeSegment<ID>(0, num_d),
-                                   RAJA::TypedRangeSegment<IG>(0, num_g),
-                                   RAJA::TypedRangeSegment<IZ>(0, num_z));
-
-  RAJA::Timer timer;
-  timer.start();
-
-  RAJA::kernel<EXECPOL>( segments,
-    [=] (IM m, ID d, IG g, IZ z) {
-       phi(m, g, z) += L(m, d) * psi(d, g, z);
-    }
-  );
+  {
+    std::cout << "\n Running RAJA sequential version of LTimes...\n";
 
-  timer.stop();
-  std::cout << "  RAJA sequential version of LTimes run time (sec.): "
-            << timer.elapsed() << std::endl;
+    std::memset(phi_data, 0, phi_size * sizeof(double));
 
-#if defined(DEBUG_LTIMES)
-  checkResult(phi, L, psi, num_m, num_d, num_g, num_z);
-#endif
-}
+    //
+    // View types and Views/Layouts for indexing into arrays
+    //
+    // L(m, d) : 1 -> d is stride-1 dimension
+    using LView = TypedView<double, Layout<2, Index_type, 1>, IM, ID>;
 
-//----------------------------------------------------------------------------//
+    // psi(d, g, z) : 2 -> z is stride-1 dimension
+    using PsiView = TypedView<double, Layout<3, Index_type, 2>, ID, IG, IZ>;
 
-{
-  std::cout << "\n Running RAJA sequential shmem version of LTimes...\n";
+    // phi(m, g, z) : 2 -> z is stride-1 dimension
+    using PhiView = TypedView<double, Layout<3, Index_type, 2>, IM, IG, IZ>;
+
+    std::array<RAJA::idx_t, 2> L_perm{{0, 1}};
+    LView L(L_data, RAJA::make_permuted_layout({{num_m, num_d}}, L_perm));
+
+    std::array<RAJA::idx_t, 3> psi_perm{{0, 1, 2}};
+    PsiView psi(psi_data,
+                RAJA::make_permuted_layout({{num_d, num_g, num_z}}, psi_perm));
+
+    std::array<RAJA::idx_t, 3> phi_perm{{0, 1, 2}};
+    PhiView phi(phi_data,
+                RAJA::make_permuted_layout({{num_m, num_g, num_z}}, phi_perm));
+
+    using EXECPOL = RAJA::KernelPolicy<statement::For<
+        0,
+        loop_exec,  // m
+        statement::For<
+            1,
+            loop_exec,  // d
+            statement::For<2,
+                           loop_exec,  // g
+                           statement::For<3,
+                                          simd_exec,  // z
+                                          statement::Lambda<0> > > > > >;
```

Comment:
Good catch!  I didn't notice that. That's disappointing since I/we exerted some effort to format the policies in the examples so they are consistent and readable.  I'll look into this to see if there's a way we can filter to avoid reformatting the policies. Thanks!